unittests/failuretests.py

```diff
@@ -373,6 +373,7 @@ def test_override_dependency_twice(self):
                                """Tried to override dependency 'foo' which has already been resolved or overridden""")
 
     @unittest.skipIf(is_windows(), 'zlib is not available on Windows')
+    @skipIfNoPkgconfig
```

Comment:
This isn't correct, there are other ways to find zlib, you can get it from cmake or it could come by our hand rolled implementation (for macos and *BSD).

I think it would be better to handle this by doing something like (the second argument is a REGEX):
```python
self.assertMesonRaises("""\
    d = dependency('zlib', required : false)'
    if not d.found():
        error('ZLIB wasn't found')
    endif
    meson.overide_dependency('zlib', declare_dependency())
    """,
    "(Tried to override dependency 'zlib' which has already been resolved or overridden|ZLIB wasn't found)")
```

Reply:
That would mean the test showing up as passed when it would be more accurate to say it was skipped, right?  Is that okay?

Reply:
We should simply drop this hunk, it's a falsehood to say that this test will error when pkg-config isn't found.

Just do not skip it.

If this test is failing on NixOS without pkg-config installed, it presumably *also* requires that zlib not be installed since otherwise the test case would, as noted, simply find it via the hand-rolled implementation.

The hand-rolled implementation is NOT "for macOS and *BSD", it is implemented everywhere and will be found everywhere, unless there is no zlib library and zlib.h header installed at all.

Reply:
Dropped.